sqlglot/dialects/dialect.py

```diff
@@ -305,7 +305,13 @@ def get_or_raise(cls, dialect: DialectType) -> Dialect:
 
             result = cls.get(dialect_name.strip())
             if not result:
-                raise ValueError(f"Unknown dialect '{dialect_name}'.")
+                from difflib import get_close_matches
+
+                similar = seq_get(get_close_matches(dialect_name, cls.classes, n=1), 0) or ""
+                if similar:
+                    similar = f" Did you mean {similar}?"
```

Comment:
A detail here is that we won't ever suggest the SQLGlot dialect `""` but that felt ok